src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/operations/diagnostics/LSPDiagnosticAnnotator.java

```diff
@@ -57,77 +54,44 @@ public LSPVirtualFileWrapper collectInformation(@NotNull PsiFile file, @NotNull
+        Document document = LSPIJUtils.getDocument(file.getVirtualFile());
+        diagnosticsPerServer.forEach(ds -> {
+            boolean codeActionsLoading = false;
+            for (Diagnostic diagnostic : ds.getDiagnostics()) {
+                codeActionsLoading = codeActionsLoading | createAnnotation(diagnostic, document, ds, holder);
             }
-        });
-
-        wrapper.setAnnotations(annotations);
-    }
-
-    private static void updateAnnotations(List<Annotation> existingAnnotations, AnnotationHolder holder) {
-        if (existingAnnotations == null || existingAnnotations.isEmpty()) {
-            return;
-        }
-        // Copy all existing annotations in the given annotation holder.
-        existingAnnotations.forEach(currentAnnotation -> {
-            // Copy range, severity, message information
-            TextRange range = new TextRange(currentAnnotation.getStartOffset() , currentAnnotation.getEndOffset());
-            Annotation newAnnotation = holder.createAnnotation(currentAnnotation.getSeverity(),  range, currentAnnotation.getMessage());
-
-            // Copy quick fixes information
-            if (currentAnnotation.getQuickFixes() == null || currentAnnotation.getQuickFixes().isEmpty()) {
-                return;
+            if (codeActionsLoading) {
+                ds.refreshQuickFixesIfNeeded();
             }
-            currentAnnotation.getQuickFixes().forEach(quickFixInfo -> newAnnotation.registerFix(quickFixInfo.quickFix));
         });
     }
 
-
-    @Nullable
-    private static Annotation createAnnotation(Document document, Diagnostic diagnostic, AnnotationHolder holder) {
+    private static boolean createAnnotation(Diagnostic diagnostic, Document document, LSPDiagnosticsForServer diagnosticsForServer, AnnotationHolder holder) {
         final int start = LSPIJUtils.toOffset(diagnostic.getRange().getStart(), document);
-        final int end = LSPIJUtils.toOffset(diagnostic.getRange().getEnd(),document);
+        final int end = LSPIJUtils.toOffset(diagnostic.getRange().getEnd(), document);
         if (start >= end) {
             // Language server reports invalid diagnostic, ignore it.
-            return null;
+            return false;
         }
+        // Create Intellij Annotation from the given LSP diagnostic
         HighlightSeverity severity = toHighlightSeverity(diagnostic.getSeverity());
         TextRange range = new TextRange(start, end);
         String message = diagnostic.getMessage();
-        return holder.createAnnotation(severity, range, message);
+        AnnotationBuilder builder = holder.newAnnotation(severity, message)
+                .range(range);
+        // Register quick fixes if there are available
+        List<IntentionAction> fixes = diagnosticsForServer.getQuickFixesFor(diagnostic);
+        boolean codeActionsLoading = LSPDiagnosticsForServer.isCodeActionsLoading(fixes);

```

Comment:
I found that these two lines should be moved above the AnnotationBuilder creation. I got some exceptions about `create()` not being called, maybe because there are some other actions in between initializing the builder and calling `create()`

Reply:
I noticed this problem while I implementing this PR. If `diagnosticsForServer.getQuickFixesFor` throws some error the annotation is registered without create which causes your problem.

I move the line code as you suggested but now diagnosticsForServer.getQuickFixesFor should never thrown some exceptions.